core/src/main/java/org/infinispan/manager/DefaultCacheManager.java

```diff
@@ -1140,7 +1140,8 @@ public ClusterExecutor executor() {
       if (transport != null) {
          long time = configurationManager.getGlobalConfiguration().transport().distributedSyncTimeout();
          return ClusterExecutors.allSubmissionExecutor(null, this, transport, time, TimeUnit.MILLISECONDS,
-               globalComponentRegistry.getComponent(ExecutorService.class, KnownComponentNames.REMOTE_COMMAND_EXECUTOR),
+               // This can run arbitrary code, including user - assume it is blocking
+               globalComponentRegistry.getComponent(ExecutorService.class, KnownComponentNames.BLOCKING_EXECUTOR),
```

Comment:
Is/should there be a way to use the non-blocking executor for users of `ClusterExecutor` that know their callback is non-blocking? Maybe internal code should be sending commands anyway?

Reply:
I was debating about this. We should probably do something though. I think having a separate JIRA is probably best. https://issues.redhat.com/browse/ISPN-11482